plugins/linux/ifplugin/interface_config.go

```diff
@@ -260,6 +264,13 @@ func (plugin *LinuxInterfaceConfigurator) DeleteLinuxInterface(linuxIf *interfac
 		peerConfig = oldConfig.peer
 	}
 
+	if _, _, exists := plugin.ifCachedConfigs.LookupIdx(linuxIf.HostIfName); exists {
+		// Unregister TAP from the in-memory map
+		plugin.log.Infof("Removing Linux Tap interface %v from cache", linuxIf.HostIfName)
```

Comment:
It is not certain here that this interface is TAP, is it? If the `ifCachedConfigs` is a cache for TAP interfaces only and those are registered in `configureTapInterface` function, then they should be unregistered inside `deleteTapInterface` function in which we are sure we work with TAP interface. Otherwise situation might arise where we delete cached TAP interface if they for some reason have same `HostIfName`.